### PR TITLE
Added dial control for reuseport

### PR DESCRIPTION
**Pull request by moredure**
This changes based on https://github.com/libp2p/go-reuseport

### CODE REVIEW

**Comment by mikefaraponov** [comment]
These functions allows to use reuseport on dial side of connection (custom dialer for fasthttp, etc.)

File: examples/client/client.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// +build go1.11
+
+package main
+
+import (
+	"fmt"
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/reuseport"
```
Comment (erikdubbelboer) [inline examples/client/client.go]: Please put the above to imports in a separate block like everywhere else ([for example](https://github.com/valyala/fasthttp/blob/master/compress.go)).

**Comment by Fenny** [comment]
Do we still want to continue with this PR? If so, we just have to change the import blocks?

**Comment by moredure** [comment]
@Fenny no need

**Pull request closed without merging**